test/test-agent-metadata.js

```diff
@@ -61,13 +57,40 @@ describe('agent interaction with metadata service', function() {
     }, 500);
   });
 
+  it('should preserve public interface when stopped', function(done) {
+      nock.disableNetConnect();
```

Comment:
It may be a better separation of concerns to break this into two tests (has same interface when disabled which manually calls `stop`, and another that ensures that the agent stops on configuration failure).